modal/shared/protocol.py

```diff
@@ -95,45 +100,29 @@ class Usage(BaseModel):
     prompt_tokens: int
     completion_tokens: int
 
-    # TODO: add these in after deprecating the old fields
-    # duration: float
-    # gpu_type: GPUType
-    # gpu_count: int
+    duration: float
+    gpu_type: GPUType
+    gpu_count: int
 
 
 class ResponseBody(BaseModel):
     text: str
-    prompt_tokens: int  # TODO: deprecate
-    completion_tokens: int  # TODO: deprecate
     usage: Usage
     done: bool
 
 
 def create_response_text(
     text: str,
-    prompt_tokens: int = 0,
-    completion_tokens: int = 0,
+    usage: Usage,
     done: bool = False,
 ) -> str:
-    return ResponseBody(
-        text=text,
-        prompt_tokens=prompt_tokens,
-        completion_tokens=completion_tokens,
-        usage=Usage(
-            prompt_tokens=prompt_tokens,
-            completion_tokens=completion_tokens,
-        ),
-        done=done,
-    ).json(ensure_ascii=False)
+    r = ResponseBody(text=text, usage=usage, done=done)
+    return r.json(ensure_ascii=False)
```

Comment:
do y'all remember if there were bugs as a result of not using ensure_ascii before? or the general motivation behind this?

was thinking of refactoring a bit since pydantic deprecated this fn recently
> The `json` method is deprecated; use `model_dump_json` instead.

Reply:
The main reason we used this was due to the pydantic version shipped with vllm - it's deprecated in latest (due to modal installation), but not if we have the vllm's pydantic typing xd.....

I wanted to do this way back in September when I saw the red deprecation strike! - Let's try it out on dev and see if it work!

Reply:
started to simplify response serialization a bit in the refactor in recent commits, but gonna save the `.json()` deprecation for a followup since enough logic is getting shuffled in this one already